sky/data/mounting_utils.py

```diff
@@ -194,9 +194,14 @@ def get_az_mount_cmd(container_name: str,
         bucket_sub_path_arg = ''
     else:
         bucket_sub_path_arg = f'--subdirectory={_bucket_sub_path}/ '
-    mount_options = '-o allow_other -o default_permissions'
+    mount_options = ['allow_other', 'default_permissions']
+    # Format: -o flag1, flag2
+    fusermount_options = '-o ' + ','.join(mount_options)
```

Comment:
minor: handle empty list if we modify it in the future?
```suggestion
    fusermount_options = '-o ' + ','.join(mount_options) if mount_options else ''
```